sc/social/like/tests/test_viewlets.py

```diff
@@ -1,125 +1,215 @@
 # -*- coding: utf-8 -*-
 from plone import api
-from plone.app.testing import setRoles
-from plone.app.testing import TEST_USER_ID
-from sc.social.like.browser.viewlets import SocialLikesViewlet
-from sc.social.like.browser.viewlets import SocialMetadataViewlet
+from profilehooks import profile
+from profilehooks import timecall
 from sc.social.like.config import IS_PLONE_5
 from sc.social.like.interfaces import ISocialLikeSettings
 from sc.social.like.testing import INTEGRATION_TESTING
+from sc.social.like.testing import load_image
+from sc.social.like.tests.api_hacks import set_image_field
 
+import contextlib
+import re
 import unittest
 
 
 do_not_track = ISocialLikeSettings.__identifier__ + '.do_not_track'
 
 
-class MetadataViewletTestCase(unittest.TestCase):
+@contextlib.contextmanager
+def capture():
+    """A context manager to capture stdout and stderr.
+    http://stackoverflow.com/a/10743550/644075
+    """
+    import sys
+    from cStringIO import StringIO
+    oldout, olderr = sys.stdout, sys.stderr
+    try:
+        out = [StringIO(), StringIO()]
+        sys.stdout, sys.stderr = out
+        yield out
+    finally:
+        sys.stdout, sys.stderr = oldout, olderr
+        out[0], out[1] = out[0].getvalue(), out[1].getvalue()
+
+
+class ViewletBaseTestCase(unittest.TestCase):
 
     layer = INTEGRATION_TESTING
 
     def setUp(self):
         self.portal = self.layer['portal']
         self.request = self.layer['request']
-        setRoles(self.portal, TEST_USER_ID, ['Manager'])
-        self.portal.invokeFactory('Document', 'my-document')
-        self.document = self.portal['my-document']
 
-    def viewlet(self, context=None):
-        context = context or self.portal
-        viewlet = SocialMetadataViewlet(context, self.request, None, None)
+        with api.env.adopt_roles(['Manager']):
+            self.obj = api.content.create(
+                self.portal, type='News Item', id='foo')
+        set_image_field(self.obj, load_image(1024, 768), 'image/png')
+
+    @property
+    def _enable_all_plugins(self):
```

Comment:
why did you use a property here? it is not returning any value nor setting up any value